app/src/main/java/com/example/concordia_campus_guide/Activities/MainActivity.java

```diff
@@ -53,11 +68,46 @@ protected void onCreate(Bundle savedInstanceState) {
 
         View infoCard = findViewById(R.id.bottom_card_scroll_view);
         swipeableInfoCard = BottomSheetBehavior.from(infoCard);
+        notification = new Notification(this,AppDatabase.getInstance(this));
+        notification.checkUpCalendarEvery5Minutes();
         showPOICard();
-
         locationFragment = (LocationFragment) getSupportFragmentManager().findFragmentById(R.id.locationFragment);
     }
 
+    public void popUp(CalendarEvent calendarEvent){
+        AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+        builder.setCancelable(true);
+        builder.setTitle("Heads up... Your next class is in " +  mViewModel.displayTimeToNextClass(calendarEvent.getStartTime()));
+        builder.setMessage("You have " + calendarEvent.getTitle() + " in " + mViewModel.displayTimeToNextClass(calendarEvent.getStartTime())  + " at " +
+                calendarEvent.getLocation() +" ! Please choose to either get directions to your class or to disregard this message");
+
+        setupCancelBtn(builder);
+        setupShowMeDirectionsBtn(builder,calendarEvent.getLocation());
+
+        builder.show();
+    }
+
+    private void setupCancelBtn(AlertDialog.Builder builder){
+        builder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                dialog.cancel();
+            }
+        });
+    }
+
+    private void setupShowMeDirectionsBtn(AlertDialog.Builder builder, String location){
+        final String locationTemp = location;
+        final Activity mainActivity = this;
+        builder.setPositiveButton("Show me Directions", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                RoomModel room = notification.getRoom(locationTemp);
```

Comment:
Can you add a check to make sure that the event is properly stored in the Calendar? When the location is not properly stored, it causes the application to crash

Reply:
Should avoid displaying the event if the location is not properly stored or how should i fix this ?
Any suggestion?

Reply:
Yes, just have a message that says that the location format is wrong in the notification